.github/workflows/memory-check.yml

```
@@ -0,0 +1,157 @@
+          components: clippy, rustfmt
+      
+      - name: Install Valgrind
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y valgrind
+      
+      - name: Install uv
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
+      
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)